### ISSUE TITLE
Get `htmx.config is undefined` error after update htmx from 1.9.12 into 2.0.2

### ISSUE THREAD

**Issue by zw963** [bigskysoftware/htmx#2909]
Following is the source code of my `js/app/js`:

```js
import 'htmx.org';
window.htmx = require('htmx.org');
import _hyperscript from 'hyperscript.org';
_hyperscript.browserInit();

// This object can be output no error.
console.log(htmx); 

// both of them not work.
document.addEventListener('htmx:load', function() {
    htmx.config.methodsThatUseUrlParams = ['get'];
});

document.addEventListener('DOMContentLoaded', function () {
    htmx.config.methodsThatUseUrlParams = ['get'];
});
```

Following screenshot is the output of console of Firefox:

![image](https://github.com/user-attachments/assets/86a6981d-16f6-4f39-9ff1-a88e09bca615)

`console.log(htmx);` output the htmx object no error, but this object seem like very different from the older 1.x version object, `htmx.config' was not defined in this case.

BTW: because following settings works both on 1.x on 2.0.2 like following:

```js
htmx.logger = function (elt, event, data) {
        if (console) {
            console.log(event, elt, data);
        }
    };
```
so, i assume i config htmx correct, , could you please help me on fix the `htmx.config` issue?

Thanks.

**Comment by MichaelWest22** [comment]
I'm not sure how to fix that issue sorry but the recommended way to set htmx.config is to use meta tags placed in your main page head. https://htmx.org/docs/#config as this is easier than trying to use event handlers and JS to do it.

**Comment by zw963** [comment]
The document said, 

> You can set them directly in javascript ...

so, i guess there maybe a way to add it directly into js.

> I'm not sure how to fix that issue sorry but the recommended way to set htmx.config is to use meta tags placed in your main page head. https://htmx.org/docs/#config as this is easier than trying to use event handlers and JS to do it.

**Comment by MichaelWest22** [comment]
Yeah you can do it with javascript just fine.  just saying it is easier to set config via meta tags and you don't have to worry about any scope issues like you have to in javascript.  To find the cause of your issue i would put a breakpoint on your code block here or add debugger;.

```javascript
document.addEventListener('htmx:load', function() {
    console.log(htmx);
    debugger;
    htmx.config.methodsThatUseUrlParams = ['get'];
});
```
You can then use browser dev tools to find if window.htmx or htmx is in scope when this event listener fires.  Maybe the location or order this event listener fires in is causing your issues.

meta tag example:
```html
<head>
<meta name="htmx-config" content='{"methodsThatUseUrlParams ":["get"]}' />
</head>
```

**Comment by zw963** [comment]
Hi, I output the htmx object like following:

```js
import 'htmx.org';
window.htmx = require('htmx.org');
import _hyperscript from 'hyperscript.org';
_hyperscript.browserInit();

document.addEventListener('htmx:load', function() {
    console.log(htmx);
    console.log(htmx.config);
});
```

please check following screenshot which is the Firefox log output:

![image](https://github.com/user-attachments/assets/08625af7-9fb5-4839-93bb-0d3754ee30f7)

The object was print in console as in the screenshot, it very different with old 1.x version htmx object, in fact, i can even can't use `htmx.onLoad` hook which I always used it before version 1.x.

I guess there is some config issue in my side, I don't familiar with the javascript, i guess this probably a import issue? because as you can see, functions like `htmx.config`, `html.onLoad` nested in the `htmx.default`, any idea?

Thanks

**Comment by zw963** [comment]
Hi, @MichaelWest22 , i figure out where is the issue come from, following code works!

```js
import htmx from 'htmx.org';
import _hyperscript from 'hyperscript.org';
_hyperscript.browserInit();

function init () {
    console.log("hello!")
    htmx.logger = function (elt, event, data) {
        if (console) {
            console.log(event, elt, data);
        }
    };
    
    htmx.config.methodsThatUseUrlParams = ['get'];
}

htmx.onLoad(init);
```

When update from 1.x into 2.x, we need change code 

From 

```js
import 'htmx.org';
window.htmx = require('htmx.org');
```

into 

```js
import htmx from 'htmx.org';
```

We need update the [webpack install doc](https://htmx.org/docs/#webpack) and [migrate 1.x to 2.x document](https://htmx.org/migration-guide-htmx-1/) to reflect this changes, right? anyway, for users like me which is a newbie of javascript, it spent lots of time to migrate.

**Comment by MichaelWest22** [comment]
#2668 can you check out the suggested updates to the documentation in this PR and see if the updated docs would help for your situation?

**Comment by zw963** [comment]
> #2668 can you check out the suggested updates to the documentation in this PR and see if the updated docs would help for your situation?

https://github.com/bigskysoftware/htmx/blob/fc272ef75fbf40e765f5941f011d7dcfaca44bec/www/content/docs.md#global-variable

I consider this doc is complicated for me, because only add ``import htmx from 'htmx.org';`` was do all trick when use with webpack,  right? move those more detailed doc for others into an new place is better?

And, for users still keep use old version htmx, i think we still need  the old 1.x import doc, right? and one more important point, we need mention this import changes in migrate doc, what do you think?

**Comment by 1cg** [comment]
resolved
